python4j/python4j-core/src/main/java/org/eclipse/python4j/PythonProcess.java

```diff
@@ -0,0 +1,127 @@
+/*******************************************************************************
+ * Copyright (c) 2020 Konduit K.K.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Apache License, Version 2.0 which is available at
+ * https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ******************************************************************************/
+
+
+package org.eclipse.python4j;
+
+import org.apache.commons.io.IOUtils;
+import org.bytedeco.javacpp.Loader;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+
+public class PythonProcess {
+    private static String pythonExecutable = Loader.load(org.bytedeco.cpython.python.class);
+    public static String runAndReturn(String... arguments)throws IOException, InterruptedException{
+        String[] allArgs = new String[arguments.length + 1];
+        for (int i = 0; i < arguments.length; i++){
+            allArgs[i + 1] = arguments[i];
+        }
+        allArgs[0] = pythonExecutable;
+        ProcessBuilder pb = new ProcessBuilder(allArgs);
+        Process process = pb.start();
+        String out = IOUtils.toString(process.getInputStream(), StandardCharsets.UTF_8);
+        process.waitFor();
+        return out;
```

Comment:
Should we redirect the error stream too - ProcessBuilder.redirectError ?

And I'm not sure (can't find a confirmation about the behaviour of the input stream), but I wonder if it's possible to have a race condition here, i.e., if process.getInputStream doesn't always block until the process terminates.

An approach like this (with StringBuilder instead of println) might be safer:
https://github.com/KonduitAI/konduit-serving/blob/9eddf78a3ddcc9c272d23e3c09785f622fed78c3/konduit-serving-cli/src/main/java/ai/konduit/serving/cli/launcher/command/ServeCommand.java#L235-L259

Reply:
Will address in new pr.